crates/sui-types/src/gas.rs

```diff
@@ -383,8 +383,14 @@ impl<'a> SuiGasStatus<'a> {
     pub fn bucketize_computation(&mut self) -> Result<(), ExecutionError> {
         let computation_cost: u64 = self.gas_used().into();
         let bucket_cost = get_bucket_cost(&COMPUTATION_BUCKETS, computation_cost);
-        let charge = bucket_cost - computation_cost;
-        self.deduct_computation_cost(&charge.into())
+        let charge = bucket_cost.saturating_sub(computation_cost);
```

Comment:
```suggestion
        // charge extra on top of `computation_cost` to make the total computation gas cost a bucket value
        let extra_charge = bucket_cost.saturating_sub(computation_cost);
```

I also think it would be useful to add a comment that says whether saturation is expected (I think not?) or this is just defensive. If it's defensive, a `debug_assert` might be appropriate.

Reply:
was this the change requested? 
done